main.go

```diff
@@ -381,18 +430,57 @@ func runCLIApp(c *cli.Context) (err error) {
 	{
 		mfs, err = mountWithArgs(bucketName, mountPoint, flags, mountConfig)
 
-		if err == nil {
+		// This utility is to absorb the error
+		// returned by daemonize.SignalOutcome calls by simply
+		// logging them as error logs.
+		callDaemonizeSignalOutcome := func(err error) {
+			if err2 := daemonize.SignalOutcome(err); err2 != nil {
+				logger.Errorf("Failed to signal to daemon: %v", err2)
+			}
+		}
+
+		markSuccessfulMount := func() {
 			// Print the success message in the log-file/stdout depending on what the logger is set to.
 			logger.Info(SuccessfulMountMessage)
+			callDaemonizeSignalOutcome(nil)
+		}
 
-			daemonize.SignalOutcome(nil)
+		if err == nil {
+			if isDynamicMount {
+				markSuccessfulMount()
+			} else {
+				switch flags.MetadataPrefetchMode {
+				case config.MetadataPrefetchModeDisabled:
+					markSuccessfulMount()
+				case config.MetadataPrefetchModeSynchronous:
+					if err = callListRecursive(mountPoint); err != nil {
+						// Printing via mountStatus will have duplicate logs on the console while
+						// mounting gcsfuse in foreground mode. But this is important to avoid
+						// losing error logs when run in the background mode.
+						logger.Errorf("%s: %v\n", UnsuccessfulMountMessagePrefix, err)
+						err = fmt.Errorf("%s: mountWithArgs: %w", UnsuccessfulMountMessagePrefix, err)
+						callDaemonizeSignalOutcome(err)
+						return
+					}
+
+					markSuccessfulMount()
+				case config.MetadataPrefetchModeAsynchronous:
+					markSuccessfulMount()
+
+					go func() {
+						if err := callListRecursive(mountPoint); err != nil {
+							logger.Errorf("Metadata-prefetch failed: %v", err)
+						}
+					}()
+				}
+			}
 		} else {
```

Comment:
We can handle the err != nil case first since it's mainly error handling and a small block of code and return. We can then handle the err == nil case and reduce which is a much more complex block and reducing its indentation will have greater readability benefits - [go/tott/651](https://goto.google.com/tott/651).

Reply:
done